Changes translate algorithm from one call for each line, to a single …

Jack-Gill-TH opened the pull request:
…call that takes an array of lines, to reduce the number of calls we're making. Whether or not this will actually reduce our utilisation, I'm not sure. It feels like it SHOULD, because we're making less calls, but we are still asking for the same amount of stuff to be translated, so it depends on how Google calculates usage.

Removes unsused function from home.
Fixes return of scrape lambda to work correctly with return handler in Result.js.

Jack-Gill-TH committed:
Changes translate algorithm from one call for each line, to a single call that takes an array of lines, to reduce the number of calls we're making.
Removes unsused function from home.
Fixes return of scrape lambda to work correctly with return handler in Result.js.

ChristienGuy commented:
Translate calculates usage by the number of characters translated. So I don't think this will make a massive difference to usage but it *will* make things much faster/more efficient overall (and looks excellent too)

What we can do to limit overall Translate usage is cache our translated results (probs somewhere like firebase)

ChristienGuy commented on src/lambda/scrape.js:
```diff
@@ -89,9 +81,11 @@ exports.handler = async (event, context) => {
     return {
         statusCode: 200,
-        body: JSON.stringify(lyricsData)
+        body: JSON.stringify(body)
```
How come you're nesting the data one more object deep? Rather than just returning lyricsData by itself?

Jack-Gill-TH replied:
Well initially it was just because I realised that Results.js was trying to access a 'lyricsData' property on the returned object, so I just fixed it on this end rather on the other - eventually it might make sense to go back to this way if we're going to return translated/untranslated lyrics, but I'll change it now in the interest of simplicity!

ChristienGuy reviewed (approved):
Looks dope, question about the data structure but nothing to prevent it merging.

Jack-Gill-TH committed:
Restructures scrape return to be more simple and appropriate.
Adds catch for specific 'too many text elements' error, and functionality to pare down what we are sending for translation, rather than just error.

Jack-Gill-TH commented:
Have a look at those changes when you get a chance @ChristienGuy - I've changed the return from scrape, but also I tried translating 'rap god' and got an error that I tried to deal with in a sensible way. See what you think, I don't know if a 'do while' has any place in a modern code base! :P

Jack-Gill-TH commented:
Google is still moaning about that JSON file being in the repo, so I'm gunna merge this now so I can have a go at filtering it out!
